internal/handlers/common/common.go

```diff
@@ -15,10 +15,31 @@
 // Package common provides common code for all handlers.
 package common
 
+import (
+	"errors"
+
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
 const (
 	// MinWireVersion is the minimal supported wire protocol version.
 	MinWireVersion = int32(0) // needed for some apps and drivers
 
 	// MaxWireVersion is the maximal supported wire protocol version.
 	MaxWireVersion = int32(17)
 )
+
+// IsNotClientMetadata checks if the message does not contain client metadata.
+func IsNotClientMetadata(msg *wire.OpMsg) error {
```

Comment:
Usually, functions called `IsSomething` mean that `bool` needs to be returned. In this case, it returns an error (and it looks to me that it makes sense as then we can return the error in the callers).
 
If we want to keep this function to return an error, I'd suggest we can name it `CheckClientMedata`.